cardano-cli/src/Cardano/CLI/EraBased/Options/Governance/DRep.hs

```diff
@@ -0,0 +1,179 @@
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE GADTs #-}
+{-# LANGUAGE LambdaCase #-}
+
+module Cardano.CLI.EraBased.Options.Governance.DRep
+  ( pGovernanceDRepCmds
+  ) where
+
+import           Cardano.Api
+
+import           Cardano.CLI.Environment
+import           Cardano.CLI.EraBased.Commands.Governance.DRep
+import           Cardano.CLI.EraBased.Options.Common
+import           Cardano.CLI.Types.Key
+
+import           Data.Foldable
+import           Data.String
+import           Options.Applicative (Parser)
+import qualified Options.Applicative as Opt
+
+pGovernanceDRepCmds :: ()
+  => EnvCli
+  -> CardanoEra era
+  -> Maybe (Parser (GovernanceDRepCmds era))
+pGovernanceDRepCmds envCli era =
+  subInfoParser "drep"
+    ( Opt.progDesc
+        $ mconcat
+          [ "DRep member commands."
+          ]
+    )
+    [ pGovernanceDRepKeyGen era
+    , pEraBasedDelegationCertificateCmd envCli era
+    , pEraBasedRegistrationCertificateCmd envCli era
+    ]
+
+pGovernanceDRepKeyGen :: ()
+  => CardanoEra era
+  -> Maybe (Parser (GovernanceDRepCmds era))
+pGovernanceDRepKeyGen era = do
+  w <- maybeFeatureInEra era
+  pure
+    $ subParser "key-gen"
+    $ Opt.info
+        ( GovernanceDRepGenerateKey w
+            <$> pVerificationKeyFileOut
+            <*> pSigningKeyFileOut
+        )
+    $ Opt.progDesc "Generate Delegate Representative verification and signing keys."
+
+-- Registration Certificate related
+
+pEraBasedRegistrationCertificateCmd :: ()
+  => EnvCli
+  -> CardanoEra era
+  -> Maybe (Parser (GovernanceDRepCmds era))
+pEraBasedRegistrationCertificateCmd envCli era = do
+  w <- maybeFeatureInEra era
+  pure
+    $ subParser "registration-certificate"
+    $ Opt.info (pEraCmd envCli w)
+    $ Opt.progDesc "Create a registration certificate."
+ where
+  pEraCmd :: EnvCli -> AnyEraDecider era -> Parser (GovernanceDRepCmds era)
+  pEraCmd envCli' = \case
+    AnyEraDeciderShelleyToBabbage sToB ->
+      GovernanceDRepRegistrationCertificateCmd
+        <$> asum [ ShelleyToBabbageStakePoolRegTarget sToB
+                     <$> pStakePoolRegistrationParserRequirements envCli'
+                 , ShelleyToBabbageStakeKeyRegTarget sToB
+                     <$> pStakeIdentifier
+                 ]
+        <*> pOutputFile
+
+    AnyEraDeciderConwayOnwards cOn ->
+      GovernanceDRepRegistrationCertificateCmd . ConwayOnwardRegTarget cOn
+        <$> asum [ RegisterStakePool cOn
+                     <$> pStakePoolRegistrationParserRequirements envCli'
```

Comment:
This is not just a DRep command

Reply:
Thanks!  Will revisit this after finding more about this requirement.